harness/determined/experimental/pytorch_lightning/adapter.py

```diff
@@ -0,0 +1,85 @@
+from typing import Callable, NewType, Any
+import pytorch_lightning as ptl
+from determined.pytorch import PyTorchTrial, PyTorchTrialContext
+from determined.experimental.pytorch_lightning.data_module import DETLightningDataModule
+from determined import monkey_patch
+from typing import Any, Dict, Sequence, Union
+from inspect import signature
+import torch
+import sys
+
+TorchData = Union[Dict[str, torch.Tensor], Sequence[torch.Tensor], torch.Tensor]
+HyperparamsProvider = Callable[[str], Any]
+
+
+def bail(msg: str = '', fail: bool = True):
+    msg = f'NotSupported: {msg}'
+    if fail:
+        raise TypeError(msg)
+    else:
+        # TODO use a logger
+        print(msg, file=sys.stderr)
+
+
+class DETLightningModule(ptl.LightningModule):
+    """
+    DETLightningModule helps us dictate what extra inputs the user's lightning module should expect.
+    Aleternatively we can avoid this and have the user take care of it.
+    """
+    def __init__(self, *args, get_hparam: HyperparamsProvider = None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.get_hparam = get_hparam
+        check_compat(self)
+
+
+def check_compat(lm: DETLightningModule):
+    if len(signature(lm.training_step).parameters) > 2:
+        bail('`optimizer_idx` and `hiddens` are not supported.')
+    if len(signature(lm.validation_step).parameters) > 2:
+        bail('`dataloader_idx` is not supported.')
+
+
+class PTLAdapter(PyTorchTrial):
```

Comment:
Same. We don't seem to need a separate class for `PTLAdapter`. Instead, we should add a function `PytorchContext.wrap_lightning_module` as in the PRD

Reply:
we updated the PRD.